lib/Dialect/FIRRTL/Transforms/SimplifyAggregates.cpp

```diff
@@ -244,12 +247,44 @@ bool MergeConnection::peelConnect(StrictConnectOp connect) {
   return true;
 }
 
-bool MergeConnection::run() {
+// Return true if multibit mux was folded into subaccess op.
+// If multbit mux inputs are `a[n-1], a[n-2], ..., a[1], a[0]`, we can fold
+// multibit into subaccess op.
+bool SimplifyAggregate::simplifyMultibitMux(MultibitMuxOp multibitMux) {
```

Comment:
Should this be a canonicalizer?

Reply:
I didn't want to create subaccess in canonicalizer because I thought most of passes assumed that subaccess was eliminated. `SimplifyAggregate` pass is intended to be ran just before LowerToHW so that other passes don't care about subaccess.